lib/Agrammon/DB/Dataset.pm6

```diff
@@ -49,58 +111,120 @@ class Agrammon::DB::Dataset does Agrammon::DB {
+
         self.with-db: -> $db {
-            my $ret = $db.query(q:to/SQL/, $value, $username, $!name, $var);
-            UPDATE data_new SET data_val = $1
+            my $ret = $db.query(q:to/SQL/, $comment, $username, $!name, $var);
+            UPDATE data_new SET data_comment = $1
              WHERE data_dataset=dataset_name2id($2,$3) AND data_var=$4
                                                        AND data_instance IS NULL
-            RETURNING data_val 
+            RETURNING data_comment
             SQL
 
             my $rows = $ret.rows;
             return $rows if $rows;
 
-            $ret = $db.query(q:to/SQL/, $value, $username, $!name, $var);
-            INSERT INTO data_new (data_dataset, data_var, data_val)
+            $ret = $db.query(q:to/SQL/, $comment, $username, $!name, $var);
+            INSERT INTO data_new (data_dataset, data_var, data_comment)
             VALUES (dataset_name2id($2,$3),$4,$1)
-            RETURNING data_val
+            RETURNING data_comment
             SQL
             $rows = $ret.rows;
             return $rows if $rows;
         }
     }
 
-    method !store-instance-variable($var, $instance, $value) {
+    method !store-instance-variable-comment($var, $instance, $comment) {
 
-        return unless $var and $value.defined and $instance;
+        return unless $var and $comment.defined and $instance;
         my $username = $!user.username;
-        
+
         self.with-db: -> $db {
-            my $ret = $db.query(q:to/SQL/, $value, $username, $!name, $var, $instance);
-            UPDATE data_new SET data_val = $1
+            my $ret = $db.query(q:to/SQL/, $comment, $username, $!name, $var, $instance);
+            UPDATE data_new SET data_comment = $1
              WHERE data_dataset=dataset_name2id($2,$3) AND data_var=$4
                                                        AND data_instance = $5
-            RETURNING data_val 
+            RETURNING data_comment
             SQL
 
             my $rows = $ret.rows;
             return $rows if $rows;
 
-            $ret = $db.query(q:to/SQL/, $value, $username, $!name, $var, $instance);
-            INSERT INTO data_new (data_dataset, data_var, data_val, data_instance)
+            $ret = $db.query(q:to/SQL/, $comment, $username, $!name, $var, $instance);
+            INSERT INTO data_new (data_dataset, data_var, data_comment, data_instance)
             VALUES (dataset_name2id($2,$3),$4,$1,$5)
-            RETURNING data_val
+            RETURNING data_comment
             SQL
             $rows = $ret.rows;
             return $rows if $rows;
         }
     }
-    
+
+    method store-input-comment($var-name, $comment) {
+        my $instance;
+
+        my $var = $var-name;
+        if $var ~~ s/\[(.+)\]/[]/ {
+            $instance = $0;
+        }
+
+        return $instance ?? self!store-instance-variable-comment($var, $instance, $comment)
+                         !! self!store-variable-comment($var, $comment);
```

Comment:
Is it correct that this passes `$var` down without removing the instance part?

Reply:
We do remove the instance name (with the regex substitution). The DB stores the variable name with the empty `[]`.